stanford_media.post_update.php

```diff
@@ -0,0 +1,355 @@
+    }
+
+    if ($caption_data = json_decode(htmlspecialchars_decode($token_element->getAttribute('data-caption')), TRUE)) {
+      $caption_text = strip_tags(check_markup($caption_data['value'], $caption_data['format']), '<a>');
+      $new_token_attributes->setAttribute('data-caption', $caption_text ?: $caption_data['value']);
+    }
+
+    $new_token = "<drupal-media$new_token_attributes></drupal-media>";
+
+    if (!empty($display_settings['linkit']['href'])) {
+      $link_attributes = new Attribute($display_settings['linkit']);
+      $new_token = "<a$link_attributes>$new_token</a>";
+    }
+    $html = str_replace($token, $new_token, $html);
+  }
+
+  return $html;
+}
+
+function _stanford_media_post_update_8202_image_display_mode($image_style) {
+  $view_modes = &drupal_static(__FUNCTION__, []);
+  if (empty($view_modes)) {
+    /** @var \Drupal\Core\Entity\Display\EntityViewDisplayInterface $display */
+    foreach (EntityViewDisplay::loadMultiple() as $display) {
+      if ($display->getTargetEntityTypeId() != 'media' || $display->getTargetBundle() != 'image') {
+        continue;
+      }
+      $field_display = $display->getComponent('field_media_image');
+      if ($field_display && isset($field_display['settings']['image_style'])) {
+        $view_modes[$field_display['settings']['image_style']] = $display->getMode();
+      }
+    }
+  }
+
+  if (!isset($view_modes[$image_style])) {
+    $image_style_entity = ImageStyle::load($image_style);
+    EntityViewMode::create([
+      'id' => 'media.stanford_image_' . $image_style,
+      'targetEntityType' => 'media',
+      'label' => $image_style_entity ? $image_style_entity->label() : $image_style,
+    ])->save();
+    /** @var \Drupal\Core\Entity\Display\EntityViewDisplayInterface $new_display */
+    $new_display = EntityViewDisplay::create([
+      'targetEntityType' => 'media',
+      'bundle' => 'image',
+      'mode' => 'stanford_image_' . $image_style,
+      'status' => TRUE,
+    ]);
+    $settings = [
+      'label' => 'hidden',
+      'type' => 'image',
+      'settings' => ['image_style' => $image_style, 'image_link' => ''],
+    ];
+    $new_display->setComponent('field_media_image', $settings);
+    $new_display->removeComponent('created');
+    $new_display->removeComponent('thumbnail');
+    $new_display->removeComponent('uid');
+    $new_display->save();
+    $view_modes[$image_style] = $new_display->getMode();
+  }
+
+  return $view_modes[$image_style];
+}
+
+/**
+ * Get the list of entities with drupal-entity tokens.
+ *
+ * @return array
+ *   List of entities needing fixed.
+ */
+function _stanford_media_post_update_8202_entity_list() {
+  $list = [];
+  module_load_install('stanford_media');
+  foreach (_stanford_media_update_8005_get_filter_fields() as $field) {
+    list($entity_type, $field_name) = explode(':', $field);
+
+    try {
+      $entity_ids = \Drupal::entityQuery($entity_type)
```

Comment:
Try/catch because it can get big?

Reply:
any time there are entity queries or entity storage loading there's a potential it could break the update.

Reply:
For future us, can you log the troublemaker then continue.

Reply:
:heavy_check_mark: